sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/ApplicationInsightsRestClient.cs

```diff
@@ -36,12 +49,23 @@ internal async Task<int> InternalTrackAsync(IEnumerable<TelemetryItem> body, Can
         /// This operation sends a blob from persistent storage that will be monitored by Azure Monitor.
         /// </summary>
         /// <param name="body">Content of blob to track.</param>
+        /// <param name="storage">Storage object</param>
         /// <param name="cancellationToken">The cancellation token to use.</param>
         /// <returns></returns>
-        internal async Task<int> InternalTrackAsync(ReadOnlyMemory<byte> body, CancellationToken cancellationToken = default)
+        internal async Task<int> InternalTrackAsync(ReadOnlyMemory<byte> body, IPersistentStorage storage, CancellationToken cancellationToken = default)
         {
             using var message = CreateTrackRequest(body);
-            await _pipeline.SendAsync(message, cancellationToken).ConfigureAwait(false);
+            try
+            {
+                await _pipeline.SendAsync(message, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
```

Comment:
This overload is to transmit the files from storage to ingestion service. If `SendAsync` fails, instead of creating the blob we should use the existing file without deleting it.